src/type.cpp

```diff
@@ -279,6 +281,17 @@ const AtomicType *AtomicType::GetAsUniformType() const {
     return asUniformType;
 }
 
+const AtomicType *AtomicType::GetAsUniformStorageType() const {
+    Assert(basicType != TYPE_VOID);
+    if (variability == Variability::Uniform)
+        return this;
+
+    asUniformType = new AtomicType(basicType, Variability::Uniform, isConst);
```

Comment:
It doesn't seem right to
(1) create new ``AtomicType`` object every single time we call ``AtomicType::GetAsUniformStorageType()`` regardless it was already initialized or not;
(2) right different types to ``asUniformType`` in ``GetAsUniformStorageType()`` and ``GetAsUniformType()``. I.e. if we call ``GetAsUniformStorageType()`` before ``GetAsUniformType()``, it will change (incorrectly) the type returned by ``GetAsUniformType()``.